### DISCUSSION TITLE
Another Notion example -- select a random item from a database

### DISCUSSION THREAD

**Discussion by dealingwith** [johnlindquist/kit#297] [Share]
Like the [last Notion example I posted](https://github.com/johnlindquist/kit/discussions/286), this one interfaces with my giant, singular Tasks database in Notion. This one selects one task at random (that does not have a status of Done or Archive).

[Install notion-random-task](https://scriptkit.com/api/new?name=notion-random-task&url=https://gist.githubusercontent.com/dealingwith/43c1079690c14481ec2e76737e5f60fc/raw/f77a17b9cfe93b0a4b8eb0f533f014aeaaadbb9f/notion-random-task.js)

```js
let notionToken = await env('NOTION_USER_TOKEN')
let databaseID = "3859b567fda3464ea5a69d5ccb56274b"

let {data} = await post(
  `https://api.notion.com/v1/databases/${databaseID}/query`,
{
  "filter": {
    "and": [
      {
        "property": "Status",
        "select": {
          "does_not_equal": "Done"
        }
      },
      {
        "property": "Status",
        "select": {
          "does_not_equal": "Archive"
        }
      }
    ]
  },
  page_size: 100
},
{
  headers: {
    Authorization: `Bearer ${notionToken}`,
    "Content-Type": "application/json",
    "Notion-Version": "2021-05-13"
  }
})

let tasks = data.results
let task = tasks[Math.floor(Math.random() * tasks.length)];
let pageID = task.id.replace(/-/g, "");
let pageURL = `https://notion.so/${databaseID}?p=${pageID}`
copy(pageURL)

await focusTab(pageURL, "Google Chrome Beta")
```

I also created one that only queries for tasks with no status (the "inbox")...

[Install notion-random-from-inbox](https://scriptkit.com/api/new?name=notion-random-from-inbox&url=https://gist.githubusercontent.com/dealingwith/24ccf1b94413168939eff0369a25bd9f/raw/019aa8bf68265fce8e9119b4a6409c1b5fc2ef7c/notion-random-from-inbox.js)

```js
let notionToken = await env('NOTION_USER_TOKEN')
let databaseID = "3859b567fda3464ea5a69d5ccb56274b"

let {data} = await post(
  `https://api.notion.com/v1/databases/${databaseID}/query`,
{
  "filter": 
  {
    "property": "Status",
    "select": {
      "is_empty": true
    }
  },
  page_size: 50
},
{
  headers: {
    Authorization: `Bearer ${notionToken}`,
    "Content-Type": "application/json",
    "Notion-Version": "2021-05-13"
  }
})

let tasks = data.results
let task = tasks[Math.floor(Math.random() * tasks.length)];
let pageID = task.id.replace(/-/g, "");
let pageURL = `https://notion.so/${databaseID}?p=${pageID}`
copy(pageURL)

await focusTab(pageURL, "Google Chrome Beta")
```

~Is there a better way to open the link in a browser? I stole this method from Ian Jones.~

**Comment by johnlindquist** [comment]
@dealingwith 

```js
await focusTab(url, "Google Chrome Beta")
```

Is probably what you're looking for.

I can't wait to play with these scripts, just need to get this next build out tomorrow so I can play a bit.

**Reply by dealingwith** [reply]
Awesome! Just updated.

**Comment by Alarid** [comment]
This is great, thanks for sharing! Do you know a way to open the page in the Notion app, instead of the browser? That would be awesome

**Reply by johnlindquist** [reply]
@Alarid You can use the `notion` file protocol to open pages. Build a string that looks like the following

```js
exec(`open "notion:/egghead/Welcome-README-f9925ed83fed4555a40058fd1dfdda93"`)
```

I _think_ with @dealingwith 's script, it would be

```js
exec(`open "notion:/${databaseID}/${pageID}"`)
```

But I haven't tested it.

**Reply by Alarid** [reply]
Awesome, fast & helpful, as always! Thanks
